tox_packet/src/messenger/conference/query_response.rs

```diff
@@ -129,47 +129,50 @@ impl PeerInfo {
 
 #[cfg(test)]
 mod tests {
+    use rand::thread_rng;
+
     use super::*;
 
     encode_decode_test!(
-        tox_crypto::crypto_init().unwrap(),
         query_response_encode_decode,
         QueryResponse::new(1, vec![
             PeerInfo {
                 peer_id: 1,
-                real_pk: gen_keypair().0,
-                temp_pk: gen_keypair().0,
+                real_pk: SecretKey::generate(&mut thread_rng()).public_key(),
+                temp_pk: SecretKey::generate(&mut thread_rng()).public_key(),
                 nickname: "1234".to_owned(),
             },
             PeerInfo {
                 peer_id: 2,
-                real_pk: gen_keypair().0,
-                temp_pk: gen_keypair().0,
+                real_pk: SecretKey::generate(&mut thread_rng()).public_key(),
+                temp_pk: SecretKey::generate(&mut thread_rng()).public_key(),
                 nickname: "56789".to_owned(),
             }])
     );
 
     #[test]
     fn peer_info_from_bytes_encoding_error() {
+        let mut rng = thread_rng();
```

Comment:
We should either create a tmp var=thread_rng everywhere or use thread_rng() in callsite. See ^ `temp_pk: SecretKey::generate(&mut thread_rng()).public_key(),`

Reply:
I used `&mut thread_rng()` inplace sometimes where it's used only once. Everywhere where it's used multiple times it's preferable to cache thread_rng call since it has overhead (not big, but still).